backend/src/test/java/com/redhat/cloud/notifications/routers/NotificationResourceTest.java

```diff
@@ -330,6 +331,114 @@ void testEventTypeFetchingByBundleApplicationAndEventTypeName() {
         assertEquals(1, eventTypes.size());
     }
 
+    @Test
+    void testEventTypeFetchingAndExcludeMutedTypes() {
+        UUID bundleId = helpers.createTestAppAndEventTypes();
+        List<Application> apps = applicationRepository.getApplications(TEST_BUNDLE_NAME);
+        Application app = apps.stream().filter(a -> a.getName().equals(TEST_APP_NAME_2)).findFirst().get();
+        UUID appId = app.getId();
+        Header identityHeader = initRbacMock(ACCOUNT_ID, ORG_ID, "user", FULL_ACCESS);
+        UUID behaviorGroupId1 = helpers.createBehaviorGroup(ACCOUNT_ID, ORG_ID, "behaviour-group-1", bundleId).getId();
+        UUID behaviorGroupId2 = helpers.createBehaviorGroup(ACCOUNT_ID, ORG_ID, "behaviour-group-2", bundleId).getId();
+
+        UUID endpointId1 = helpers.createWebhookEndpoint(ACCOUNT_ID, ORG_ID);
+        UUID endpointId2 = helpers.createWebhookEndpoint(ACCOUNT_ID, ORG_ID);
+        List<EventType> eventTypes = applicationRepository.getEventTypes(appId);
+        // bgroup1 assigned to ev0 and ev1, bgroup2 assigned to ev14, all other event types unassigned
+        List<String> unmutedEventTypeNames = List.of(String.format(TEST_EVENT_TYPE_FORMAT, 0), String.format(TEST_EVENT_TYPE_FORMAT, 1), String.format(TEST_EVENT_TYPE_FORMAT, 14));
+        behaviorGroupRepository.updateEventTypeBehaviors(ORG_ID, eventTypes.stream().filter(ev -> ev.getName().equals(unmutedEventTypeNames.getFirst())).findFirst().get().getId(), Set.of(behaviorGroupId1));
+        behaviorGroupRepository.updateEventTypeBehaviors(ORG_ID, eventTypes.stream().filter(ev -> ev.getName().equals(unmutedEventTypeNames.get(1))).findFirst().get().getId(), Set.of(behaviorGroupId1));
+        behaviorGroupRepository.updateBehaviorGroupActions(ORG_ID, behaviorGroupId1, List.of(endpointId1));
```

Comment:
While adding actions to the behavior groups makes sense, it's not strictly needed in the tests you wrote because the `behavior_group_action` table from the DB has no impact on the list of event types returned when `excludeMutedTypes` is set to `true`. In other words, you could skip all `*updateBehaviorGroupActions*` instructions as well as the webhook endpoints creation in the tests added in this PR.